### DISCUSSION TITLE
Example of ViewModel with constructor parameters.

### DISCUSSION THREAD

**Discussion by pablichjenkov** [ComposeGears/Tiamat#19] [General]
I see the ViewModel example uses a non-parameter constructor. Could be possible share an example of a ViewModel with dependencies as constructor params.

**Comment by egorikftp** [comment]
Hello, @pablichjenkov 
Recently we have released additional artifact with koin integration and added sample with dynamic parameter from navArgs

https://github.com/ComposeGears/Tiamat/blob/dc108ee8ae13d73e4d67ac5c6045e94d1b92f08b/example/composeApp/src/commonMain/kotlin/content/examples/koin/KoinIntegration.kt#L85

**Reply by pablichjenkov** [reply]
Ah great I didn't check that one. Thanks

**Reply by vkatz** [reply]
In addition you may pass params directly into viewModel constructor
```kotlin
class ViewModelsRootViewModel(val param: Int) : TiamatViewModel() {
   ...
}

val ViewModelsRoot by navDestination<SomeArgs> {
    val navController = navController()
    val args = navArgs()
    val viewModel = rememberViewModel { ViewModelsRootViewModel(args.param) } //<--
    ...    
}
```

**Reply by pablichjenkov** [reply]
Yes I know, using navArgs is another alternative but I meant for the case where the ViewModel is created outside the Composable code.
So the koin example is more aligned with what I am looking for.
